app/models/manageiq/providers/redfish/inventory/collector/physical_infra_manager.rb

```diff
@@ -1,19 +1,40 @@
 module ManageIQ::Providers::Redfish
   class Inventory::Collector::PhysicalInfraManager < Inventory::Collector
+    def collect
+      physical_servers
+      physical_racks
+      physical_chassis
+      firmware_inventory
+    ensure
+      disconnect!
+    end
+
     def physical_servers
-      rf_client.Systems.Members
+      @physical_servers ||= connection.Systems.Members
     end
 
     def physical_racks
-      rf_client.Chassis.Members.select { |c| c.ChassisType == "Rack" }
+      @physical_racks ||= connection.Chassis.Members.select { |c| c.ChassisType == "Rack" }
     end
 
     def physical_chassis
-      rf_client.Chassis.Members.reject { |c| c.ChassisType == "Rack" }
+      @physical_chassis ||= connection.Chassis.Members.reject { |c| c.ChassisType == "Rack" }
     end
 
     def firmware_inventory
-      rf_client.UpdateService.FirmwareInventory&.Members || []
+      @firmware_inventory ||= connection.UpdateService.FirmwareInventory&.Members || []
+    end
+
+    private
+
+    def connection
+      @connection ||= manager.connect
+    end
```

Comment:
Move the connection to being private (from [here](https://github.com/ManageIQ/manageiq-providers-redfish/pull/114/files#diff-93fa07c1a4e5a9061b02c80ac7dcc5b1L5)) to make sure the parser doesn't do any API calls directly thus bypassing the logout